test/attr/attr_availability_transitive_osx.swift

```diff
@@ -35,3 +39,43 @@ func osx_extension_call_osx_extension() {
 func osx_extension_call_osx() {
     osx() // expected-error {{'osx()' is unavailable}}
 }
+
+@available(OSX, unavailable)
+var osx_init_osx = osx() // OK
+
+@available(OSXApplicationExtension, unavailable)
+var osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
```

Comment:
Would this test pass?
```swift
@available(OSX, unavailable)
var osx_inner_init_osx = { let inner_var = osx() }
```
If so, is it worth adding to the test suite? If not, do we want to merge this change anyway?

Reply:
Definitely seems worth adding! I think it'll work with this change, but it's always worth checking.

Reply:
Aha, it does *not* work. Important info!